File: FrontEnd/GestionRA-FrontEnd/src/app/Rubricas/Modelos/rubrica.ts
import { Criterio } from "./criterio";

export class Rubrica {

   idCurso!: string;
   descripcion!: string;   
   ponderado!: number;
   idRaAsignatura!: string;
   criterios!: Criterio[];

   public constructor() {
      this.descripcion = '';
      this.ponderado = 0;
      this.idCurso = '';
      this.idRaAsignatura = '';
      this.criterios = [];
   }
}

File: BackEnd/servicioRubricas/src/main/java/co/unicauca/ra/servicio_rubricas/capaAcesoAdatos/model/Rubrica.java
package co.unicauca.ra.servicio_rubricas.capaAcesoAdatos.model;

import java.util.List;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 *
 * @author ashle
 */
@Document(collection = "rubricas")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Rubrica {
    @Id
    private String id;

    private String idCurso; 
    private String descripcion;
    private float ponderado; 
    private String idRaAsignatura;
    private List<Criterio> criterios;
}
